shell/platform/darwin/common/availability_version_check.h

```diff
@@ -0,0 +1,22 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <cstdint>
+#include <optional>
+
+namespace flutter {
+
+struct ProductVersion {
```

Comment:
Consider using `using ProductVersion = std::tuple<int32_t /* major */, int32_t /* minor */, int32_t /* patch */>;` instead. It is shorter, but more importantly, there is a handy default lexicographical compare (`>=`, `<=`) that you won't need to reinvent like you did above.

Reply:
Done. Thanks for the suggestion!